asn1/InterledgerProtocol.asn

```diff
@@ -27,4 +28,21 @@ InterledgerProtocolPaymentMessage ::= SEQUENCE {
     }
 }
 
+InterledgerProtocolError ::= SEQUENCE {
+    -- Standardized error code
+    code IA5String (SIZE (3)),
+    -- Corresponding error name
+    codeName IA5String,
+    -- Participant that originally emitted the error
+    triggeredBy Address,
+    -- Participants that forwarded the error
+    -- TODO: This could be more trouble than it's worth - could get really big etc.
```

Comment:
Great observation. I think it would be good for us to have a traceroute mechanism. If we can repurpose some existing mechanism and not have to create a dedicated one that would be awesome.

Just like IP traceroute, if you don't want people to be able to traceroute your address, you can leave this field empty when generating errors or not generate an error.